packages/app/src/systems/Chains/eth/hooks/useManageEthAssets.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react';
+import { isAddress } from 'viem';
+import { useToken } from 'wagmi';
+
+import { useAssets } from './useAssets';
+
+export const useManageEthAssets = (newAssetAddress: string) => {
+  const { assets, handlers } = useAssets();
+
+  const { data, isError, isLoading } = useToken({
+    address: newAssetAddress as `0x${string}`,
```

Comment:
can we skip searching if it's not a valid address? like:

`isAddress(newAssetAddress) ? newAssetAddress : ''`